openstack/compute/v2/extensions/evacuate/requests.go

```diff
@@ -0,0 +1,51 @@
+package evacuate
+
+import (
+	"github.com/gophercloud/gophercloud"
+)
+
+// EvacuateOptsBuilder allows extensions to add additional parameters to the
+// the Evacuate request.
+type EvacuateOptsBuilder interface {
+	ToEvacuateMap() (map[string]interface{}, error)
+}
+
+// EvacuateOpts specifies Evacuate action parameters.
+type EvacuateOpts struct {
+	// The name of the host to which the server is evacuated
+	Host string `json:"host,omitempty"`
+
+	// Indicates whether server is on shared storage
+	OnSharedStorage bool `json:"onSharedStorage"`
+
+	// An administrative password to access the evacuated server
+	AdminPass string `json:"adminPass,omitempty"`
+}
+
+// ToServerGroupCreateMap constructs a request body from CreateOpts.
+func (opts EvacuateOpts) ToEvacuateMap() (map[string]interface{}, error) {
+	return gophercloud.BuildRequestBody(opts, "evacuate")
+}
+
+// Evacuate will Evacuate a failed instance to another host.
+func Evacuate(client *gophercloud.ServiceClient, id string, opts EvacuateOptsBuilder) (r EvacuateResult) {
+	b, err := opts.ToEvacuateMap()
+	if err != nil {
+		r.Err = err
+		return
+	}
+	_, r.Err = client.Post(actionURL(client, id), b, &r.Body, &gophercloud.RequestOpts{
+		OkCodes: []int{200},
+	})
+	return
+}
+
+func (r EvacuateResult) ExtractAdminPass() (string, error) {
+	if r.Err != nil && r.Err.Error() != "EOF" {
+		return "", r.Err
+	}
+	if l, ok := r.Header["adminPass"]; ok && len(l) > 0 {
+		return l[0], nil
+	}
+	return "", nil
+}
```

Comment:
Nice work - this is really close. What you've done is correct, but we want to change it to utilize some core Gophercloud methods that already exist.

See [here](https://github.com/gophercloud/gophercloud/blob/master/openstack/compute/v2/servers/results.go#L138-L144) for a similar method to use an example. You will still want to check for the EOF error in the same way, but first extract `adminPass` like how it's done in the example.